apps/src/p5lab/actions.js

```diff
@@ -45,6 +46,9 @@ export function changeInterfaceMode(interfaceMode, spritelabDraw) {
       dispatch(show(Goal.NEW_ANIMATION));
       dispatch(pickNewAnimation());
     }
+    dispatch(
+      setAllowInstructionsResize(interfaceMode === GameLabInterfaceMode.CODE)
+    );
```

Comment:
Should we just allow `redux/instructions.js` to respond to the `CHANGE_INTERFACE_MODE` action?

Pros: We push some more business logic down into redux.  We don't introduce a new redux action that might not make sense to dispatch in isolation.

Cons: We entangle two redux modules?  But not much, because the `CHANGE_INTERFACE_MODE` action is defined in this file, I think.
Con:

Reply:
I like where you're going with this, but I don't think we want to do exactly what you're suggesting because it doesn't make sense for the instructions redux to know which modes it should be visible in. For example, instructions are visible in applab when not in code mode.

If we wanted to improve our separation of concerns, one idea is that the `GameLabView` could pass the `isVisible` prop on `InstructionsWithWorkspace`. `InstructionsWithWorkspace` would then use that information to set the `allowResize` prop on `TopInstructions` (and possibly also to `CodeWorkspaceContainer` if we someday want to make it stop triggering resize events).

one nuance is that I'm suggesting `allowResize` over `hidden` or `visible` in part because `TopInstructions` already has a `hidden` prop which means something different -- i.e. set height to zero, as opposed to temporarily disallow changing the height.

Does this seem worth doing now?

Reply:
I see what you're saying. I think it's fine to use your current solution and unblock starlabs.  Thanks for talking through alternatives!

Reply:
thinking on this a moment longer, there are 3 steps that would take us from the solution in this PR to the one I'm describing in the above comments:
1. move responsibility for communicating with instructions from p5lab redux to GameLabView
2. move understanding that "instructions are not visible" means "do not allow instructions resize" out of p5lab and into instructions
3. change allowResize from redux to an explicitly passed prop

if we want to move in this direction, let's be deliberate about which of the above are important. if we only care about (2), for example, then we might only need to rename `setAllowInstructionsResize` to `setIsInstructionsVisible`.

Reply:
sorry I missed your comment from a couple minutes ago. ok, merging as-is sounds good!